src/mysql.c

```diff
@@ -973,6 +977,31 @@ static int mysql_read (user_data_t *ud)
 				counter_submit ("mysql_innodb_rows", "read", val, db);
 			else if (strcmp (key, "Innodb_rows_updated") == 0)
 				counter_submit ("mysql_innodb_rows", "updated", val, db);
+
+ 			/* checkpoints */
+ 			else if (strcmp (key, "Innodb_checkpoint_age") == 0)
+ 				gauge_submit ("mysql_innodb_checkpoint", "age", val, db);
+ 			else if (strcmp (key, "Innodb_checkpoint_max_age") == 0)
+ 				gauge_submit ("mysql_innodb_checkpoint", "max_age", val, db);
+ 
+ 			/* max id */
+ 			else if (strcmp (key, "Innodb_max_trx_id") == 0)
+ 				gauge_submit ("mysql_innodb_max", "trx_id", val, db);
+ 
+ 			/* page_size */
+ 			else if (strcmp (key, "Innodb_page_size") == 0)
+ 				gauge_submit ("mysql_innodb_page", "size", val, db);
+ 
+ 			/* open tables and files */
+ 			else if (strcmp (key, "Innodb_open_files") == 0)
+ 				gauge_submit ("mysql_innodb_open", "files", val, db);
+ 			else if (strcmp (key, "Innodb_open_tables") == 0)
+ 				gauge_submit ("mysql_innodb_open", "tables", val, db);
```

Comment:
The type *mysql_innodb_open* does not exist. I propose to use the *objects* type and set *type_instance* to "tables-open".